Update Parameter bounds when changing units

israelmcmc opened the pull request:
Closes #92 

This performs the same conversion for `min_value`/`max_value` as is was already being done for `value`.

israelmcmc commented:
Well, it seems this conflicts with something in the test suite. I'll check what's going on.

israelmcmc commented:
I think I fixed the tests that were failing due to my changes, and the remaining ones failing are unrelated. @ndilalla can you take a look, please?

ndilalla commented:
Thanks @israelmcmc! I will review both PRs today and merge them to the dev branch. Then I can tag a new dev version `2.5.0.dev3` which will be shipped with the upcoming stable release of `astromodels 2.5.0`

israelmcmc commented:
Thanks @ndilalla!

ndilalla commented:
@israelmcmc Do you think you can add a very simple test for this PR? Like the one you posted on the issue page. To make sure we test this feature from now on

israelmcmc commented:
Thanks for the suggestion @ndilalla, makes sense. I added a test. While I was working on the test, I also realized that it is critical that the bounds are updated first, before updating the value. I fixed that as well.

israelmcmc commented:
Thanks @ndilalla!
